adding reference tests for protectionsState (#112)
duckduckgo/privacy-reference-tests: broken-site-reporting/README.md

- `remoteConfigEtag` - string - string representation of remote configuration etag
- `remoteConfigVersion` - string - string representation of remote configuration version (note, this is the numeric version found in the remote config (e.g, `1680178584671`, not `v1` or `v2`))
- `providedDescription` - string - user-provided breakage description

> - `protectionsEnabled` - boolean - if protections are enabled (true) or disabled (false). Reasons for being disabled could be:
>     - current domain exists in the user's allow-list
>     - current domain exists in the 'unprotectedTemporary' section of the privacy-configuration
>     - current domain is an exception for the `contentBlocking` feature in the privacy-configuration
>     - the feature `contentBlocking` is disabled via privacy-configuration

Reviewer: ```suggestion
- `protectionsEnabled` - boolean - if protections are enabled (true) or disabled (false) as visible to the user on the dashboard.
```
I think we can drop the reasons from here. Those are explained in the spec and it will be easier to keep them updated if we have them in one central place.

"As visible" part should make it clear that the whole "reasons for being disabled" logic applies.
